packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -25,14 +81,10 @@ export const ownSignatureKeyPaths = async (
     ).filter(isNotNil)
   ).map(({ type, index }) => ({ index, role: Number(type) }));
 
-  const rewardAccounts = new Set(knownAddresses.map(({ rewardAccount }) => rewardAccount));
+  const rewardAccounts: Account[] = knownAddresses.map(({ rewardAccount, stakeKeyDerivationPath }) => ({
```

Comment:
Could save some iterations over certificates
```suggestion
import uniqBy from 'lodash/uniqBy';
...
  const rewardAccounts: Account[] = uniqBy(knownAddresses.map(({ rewardAccount, stakeKeyDerivationPath }) => ({
  ...
  })), 'rewardAccount')
```